plugin/core/src/main/groovy/com/novoda/gradle/release/internal/AndroidAttachments.groovy

```diff
@@ -3,19 +3,23 @@ package com.novoda.gradle.release.internal
 import com.novoda.gradle.release.MavenPublicationAttachments
 import org.gradle.api.Project
 import org.gradle.api.Task
+import org.gradle.api.component.SoftwareComponent
 import org.gradle.api.tasks.javadoc.Javadoc
 
 class AndroidAttachments extends MavenPublicationAttachments {
 
+    private static final String ANDROID_SOFTWARE_COMPONENT_COMPAT_4_1 = 'com.novoda.release.internal.compat.gradle4_1.AndroidSoftwareComponentCompat_Gradle_4_1'
+
     AndroidAttachments(String publicationName, Project project, def variant) {
         super(androidComponentFrom(project),
                 androidSourcesJarTask(project, publicationName, variant),
                 androidJavadocsJarTask(project, publicationName, variant),
                 androidArchivePath(variant))
     }
 
-    private static AndroidSoftwareComponent androidComponentFrom(Project project) {
-        return new AndroidSoftwareComponent(project.objects, project.configurations)
+    private static SoftwareComponent androidComponentFrom(Project project) {
+        def clazz = this.classLoader.loadClass(ANDROID_SOFTWARE_COMPONENT_COMPAT_4_1)
```

Comment:
Right now we are not checking the current version of Gradle at runtime, we will once we have more compat implementations.